### PR TITLE
feat: update recommendations page design

### PR DESCRIPTION
**Pull request by syedsajjadkazmii**
VAN-1598

### Description

Update Post registration recommendations design according to [Figma](https://www.figma.com/file/nDYPI1KPAtzEOXAwESPlgW/Post-completion-recommendations?node-id=203%3A1169&mode=dev)

#### JIRA

[VAN-1598](https://2u-internal.atlassian.net/browse/VAN-1598)

#### How Has This Been Tested?

locally

#### Screenshots/sandbox (optional):

|Screen size|Loading state|Recommendations|
|---------|----------------|------------------|
|X Large|Screenshot 2023-08-16 at 10 15 49 AM|Screenshot 2023-08-16 at 10 18 40 AM|
|Large|Screenshot 2023-08-16 at 10 17 03 AM|
|Medium|Screenshot 2023-08-16 at 10 16 24 AM|Screenshot 2023-08-16 at 10 18 57 AM|
|Tab|Screenshot 2023-08-16 at 10 16 36 AM|
|Small|Screenshot 2023-08-16 at 10 16 12 AM|Screenshot 2023-08-16 at 10 19 15 AM|

### CODE REVIEW

File: src/recommendations/ProductCard/index.jsx [hunk 1]
```diff
@@ -82,8 +83,10 @@ const ProductCard = ({
       productTypeCopy={productTypeCopy}
       productType={productType}
       variant={variant}
+      isLoading={isLoading}
       footer={(
         <Footer
+          isLoading={isLoading}
```
Comment (mubbsharanwar) [inline src/recommendations/ProductCard/index.jsx:89]: Why do we need `isLoading` in `Footer`?

File: src/recommendations/RecommendationsList.jsx [hunk 2]
```diff
@@ -5,19 +5,19 @@ import PropTypes from 'prop-types';
...
         recommendations.map((recommendation, idx) => (
-          <span key={recommendation.uuid}>
-            <ProductCard
-              product={recommendation}
-              position={idx}
-              userId={userId}
-            />
-          </span>
+          <ProductCard
+            key={recommendation.uuid}
```
Comment (mubbsharanwar) [inline src/recommendations/RecommendationsList.jsx:15]: Are we getting `uuid` from Algolia index?

File: src/recommendations/RecommendationsPage.jsx [hunk 3]
```diff
@@ -21,8 +27,10 @@ const RecommendationsPage = ({ location }) => {
   const educationLevel = EDUCATION_LEVEL_MAPPING[location.state?.educationLevel];
   const userId = location.state?.userId;
 
+  const isExtraSmall = useMediaQuery({ maxWidth: breakpoints.extraSmall.maxWidth - 1 });
```
Comment (mubbsharanwar) [inline src/recommendations/RecommendationsPage.jsx:30]: Is this only for extra small devices? Should it not be for small devices too? is this mentioned that the design should be different only for **extra small** devices?

File: src/recommendations/RecommendationsPage.jsx [hunk 4]
```diff
@@ -54,62 +62,60 @@ const RecommendationsPage = ({ location }) => {
...
+          )}
+          {isExtraSmall && (
+            <RecommendationsSmallLayout
+              userId={userId}
+              isLoading={isLoading}
+              personalizedRecommendations={algoliaRecommendations}
+              trendingProducts={trendingProducts}
+              popularProducts={popularProducts}
+            />
+          )}
```
Comment (mubbsharanwar) [inline src/recommendations/RecommendationsPage.jsx:101]: Can we use ternary operator for better readability?

File: src/recommendations/data/loadingCoursesPlaceholders.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  queryID: 'skeleton',
+  recentEnrollmentCount: 0,
+  title: 'skeleton',
+  topics: [{
+    topic: 'skeleton',
+  }],
+  uuid: 'skeleton',
+};
+
+const loadingCoursesPlaceHolders = [
```
Comment (mubbsharanwar) [inline src/recommendations/data/loadingCoursesPlaceholders.js:29]: In trending there are 3 products. is it ok if we show the skeleton for 4 products?

**Review by mubbsharanwar — APPROVED** [review]
Looks, good with some queries.

File: src/recommendations/ProductCard/index.jsx (hunk 1 above)
Comment (syedsajjadkazmii) [reply src/recommendations/ProductCard/index.jsx:89]: oh that was for some random testing, i will remove it.

File: src/recommendations/RecommendationsList.jsx (hunk 2 above)
Comment (syedsajjadkazmii) [reply src/recommendations/RecommendationsList.jsx:15]: yes we are

File: src/recommendations/RecommendationsPage.jsx (hunk 3 above)
Comment (syedsajjadkazmii) [reply src/recommendations/RecommendationsPage.jsx:30]: yes as per figma, its for extra small devices

File: src/recommendations/data/loadingCoursesPlaceholders.js (hunk 5 above)
Comment (syedsajjadkazmii) [reply src/recommendations/data/loadingCoursesPlaceholders.js:29]: yep. this is what the design says.

**Pull request merged**